### ISSUE TITLE
'usb timeout error' when used on a raspberry pi 3 with a cheap POS-58 thermal printer

### ISSUE THREAD

**Issue by BimBim134** [python-escpos/python-escpos#628]
I have:
- [x] searched open and closed issues for duplicates

### Bug description
hi there, **first ever** Issue submission on GitHub so be kind ;)

I'm trying to make a program that can print images and text on my cheap printer but when I do so, **it print the first few line, raise an error and then proceed to print complete garbage**.

#### error
here is the error i get everytime:
```
The media.width.pixel field of the printer profile is not set. The center flag will have no effect.
Traceback (most recent call last):
  File "/home/bimbim/codes/printer/main.py", line 55, in <module>
    main()
  File "/home/bimbim/codes/printer/main.py", line 52, in main
    p._raw(d.output)
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/escpos/printer.py", line 99, in _raw
    self.device.write(self.out_ep, msg, self.timeout)
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/usb/core.py", line 989, in write
    return fn(
           ^^^
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/usb/backend/libusb1.py", line 837, in bulk_write
    return self.__write(self.lib.libusb_bulk_transfer,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/usb/backend/libusb1.py", line 938, in __write
    _check(retval)
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/usb/backend/libusb1.py", line 602, in _check
    raise USBTimeoutError(_strerror(ret), ret, _libusb_errno[ret])
usb.core.USBTimeoutError: [Errno 110] Operation timed out
```

#### program
here is the program I wrote so far (still a WIP):
```python
#!/usr/bin/env python

from glob import glob
from pathlib import Path
import sys
from time import sleep
import textwrap

from escpos import config
from escpos.printer import Usb, Dummy

def get_printer():
       p = Usb(idVendor=0x456,
               idProduct=0x0808,
               in_ep=0x81,
               out_ep=0x03,
               timeout=800,
               profile='POS-5890')
       print('usb printer loaded\n')
       return p

def argParse(arg):
    file = Path(arg)
    imgFormat = ['.jpg', '.jpeg', '.png']
    textFormat = ['.txt']
    if file.suffix in imgFormat:
        return ('img', file.resolve())
    elif file.suffix in textFormat:
        with open(file, 'r') as txt:
            return ('text', txt.read())
    else:
        return f"not a valid file to be printed"

def main():
    d = Dummy()
    p = get_printer()
    for a in sys.argv:
        obj = argParse(a)
        if obj[0] == 'text':
            wrapper = textwrap.TextWrapper(width=32)
            output = wrapper.wrap(obj[1])
            print('text:')
            for line in output:
                print(f'\t{line}')
                d.textln(line)

        elif obj[0] == 'img':
            print(f'image:\n- {a}\n')
            d.image(obj[1])
    p._raw(d.output)

if __name__=="__main__":
    main()
```

*please take note the error is there even if I use `p.text` or `p.image` directly instead of using `p._raw(d.output)`*

### Steps to reproduce
- install the necessary library
- use the program above
- run it like so `./main.py your_text.txt your_image.jpg`

### Device info

 **Printer:** [HOIN DNT-585-LU2511](https://www.hoinprinter.com/products/details/50.html)

 **python-escpos version:** 2.0.0

 **python version:**  3.11.2

 **operating system:**
 - Operating System: Debian GNU/Linux 12 (bookworm)  
 - Kernel: Linux 6.1.0-rpi7-rpi-v8
 - Architecture: arm64

**hardware:** raspberry pi 3

**Comment by belono** [comment]
Hello @BimBim134  and welcome to python-escpos.

We hope we can help you with your first issue submission :1st_place_medal: .

Before we start it is preferable that we are sure that your environment is correctly set up, so please, follow the [installation guide](https://python-escpos.readthedocs.io/en/latest/user/installation.html) to assert you are using the latest (v.3.0) python-escpos version and all the required dependencies. Also remember that to print via the USB bus in python on Linux you need to setup an [udev rule](https://python-escpos.readthedocs.io/en/latest/user/installation.html#setup-udev-for-usb-printers) to give access permission to the python scrips to the USB device.

After the environment is correctly set up:
Please, try printing some single text.
Then, if all goes well, try to send an small image to the printer. That's an image that fits the paper width.

What's the output?

**Comment by BimBim134** [comment]
Hi Belono ! thanks for your help :grin: 

I somehow missed the installation guide so I followed every step described.

**I tested with a little text (just a few lines) and it went allright** so I sent an image but then again, I got the timeout error and the garbage printed :


_note : I used my script (easyprint) to send a text, an image, another text and another image_
```
❯ easyprint 00.txt resized-0_sierra.png 01.txt resized-1_sierra.png
usb printer loaded

text:
	Un jour normal comme un autre,
	ordinaire, fait pareil que
	d'habitude et tout, elle me
	tombe dessus. Il parait que ça
	faisait 6 ans. Je me souvenais
	bien, alors on a papotté.
image:
- resized-0_sierra.png

The media.width.pixel field of the printer profile is not set. The center flag will have no effect.
text:
	Vraiment, on rigolait bien.
image:
- resized-1_sierra.png

The media.width.pixel field of the printer profile is not set. The center flag will have no effect.
Traceback (most recent call last):
  File "/home/bimbim/codes/printer/main.py", line 55, in <module>
    main()
  File "/home/bimbim/codes/printer/main.py", line 52, in main
    p._raw(d.output)
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/escpos/printer/usb.py", line 190, in _raw
    self.device.write(self.out_ep, msg, self.timeout)
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/usb/core.py", line 989, in write
    return fn(
           ^^^
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/usb/backend/libusb1.py", line 837, in bulk_write
    return self.__write(self.lib.libusb_bulk_transfer,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/usb/backend/libusb1.py", line 938, in __write
    _check(retval)
  File "/home/bimbim/codes/printer/.venv/lib/python3.11/site-packages/usb/backend/libusb1.py", line 602, in _check
    raise USBTimeoutError(_strerror(ret), ret, _libusb_errno[ret])
usb.core.USBTimeoutError: [Errno 110] Operation timed out
```
here's a picture of what it printed if that helps :
![IMG_20240223_233940](https://github.com/python-escpos/python-escpos/assets/16063801/befdda05-b577-4f29-a0c7-d663ae9b3a3f)

so as you can see, it began to print correctly and then proceed to not print what it's suppose to as if it overflow a buffer inside the printer or something like that.
if that make sense...

**Comment by BimBim134** [comment]
Hi Benolo !

`bitImageColumn` was the one !

I'm **SO** glad it works you made my day !!

note : you need to push the `timeout` parameter in the printer configuration way higher (like 5000 or 10000 ms) to be able to print for longer time in one go. Also, you need to push the `fragment_height` in the `p.image()` function also higher (like 5000px) so it don't split the image (because it crash the printer...).

Thank you so much Benolo and happy coding :grin:

**Comment by belono** [comment]
Great!!!

Thanks to you too for sharing the tip on timeout and fragment_height. I'm sure it will be helpful to other users.

Feel free to open a new issue if you encounter any other problems.

Cheers! :beer:
